src/lib.rs

```diff
@@ -1223,4 +1223,70 @@ mod test {
         let result = select_coin(&inputs, options);
         assert!(matches!(result, Err(SelectionError::InsufficientFunds)));
     }
+
+    #[test]
+    fn test_select_coin_knapsack() {
+        let inputs = vec![
+            OutputGroup {
+                value: 1000,
+                weight: 1,
+                creation_sequence: Some(0),
+                input_count: 1,
+                is_segwit: false,
+            }, // Small UTXO
+            OutputGroup {
+                value: 2000,
+                weight: 2,
+                creation_sequence: Some(1),
+                input_count: 1,
+                is_segwit: false,
+            }, // Medium UTXO
+            OutputGroup {
+                value: 3000,
+                weight: 3,
+                creation_sequence: Some(2),
+                input_count: 1,
+                is_segwit: false,
+            }, // Large UTXO
+            OutputGroup {
+                value: 4000,
+                weight: 4,
+                creation_sequence: Some(3),
+                input_count: 1,
+                is_segwit: false,
+            }, // Extra Large UTXO
+        ];
+
+        let target_value = 5000;
+
+        let options = CoinSelectionOpt {
+            target_value,
```

Comment:
because the input values are static, it doesn't serve any purpose to declare a separate variable target_value outside the struct.